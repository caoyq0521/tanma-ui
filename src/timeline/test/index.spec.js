import { mount } from '@vue/test-utils';
import TmTimeline from '../timeline.vue';

describe('tmTimeline', () => {
  it('render timeline', () => {
    const wrapper = mount(TmTimeline);
    expect(wrapper).toMatchSnapshot();
  });

  it('create', () => {
    const wrapper = mount({
      template: `<tm-timeline>
        <tm-timeline-item
          v-for="(item, index) in activities" 
          :timestamp="item.timestamp"
          :key="index"
        >
        {{ item.content }}
        </tm-timeline-item>
      </tm-timeline>`,
      data() {
        return {
          activities: [
            {
              content: '活动按期开始',
              timestamp: '2018-04-15'
            }, 
            {
              content: '通过审核',
              timestamp: '2018-04-13'
            }, 
            {
              content: '创建成功',
              timestamp: '2018-04-11'
            }
          ]
        }
      }
    })

    console.log(wrapper.vm.activities);
    const contentElms = wrapper.findAll('.tm-timeline-item__content').at(0);
    console.log("caoyq0521 ~ file: index.spec.js ~ line 43 ~ it ~ contentElms", contentElms.html())
    // contentElms.forEach((elm, index) => {
    //   console.log(elm);
    //   // expect(elm.innerText).to.equal(vm.activities[index].content);
    // });

    // console.log(wrapper.findAll('.tm-timeline-item__content'));
  })
})
  