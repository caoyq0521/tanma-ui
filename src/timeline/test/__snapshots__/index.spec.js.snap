// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tmTimeline render timeline 1`] = `<div class="tm-timeline"></div>`;
